flytekit/experimental/eager_function.py

```diff
@@ -179,7 +179,11 @@ async def __call__(self, **kwargs):
         self.async_stack.set_node(node)
 
         poll_interval = self._poll_interval or timedelta(seconds=30)
-        time_to_give_up = datetime.max if self._timeout is None else datetime.now(timezone.utc) + self._timeout
+        time_to_give_up = (
+            (datetime.max - timedelta(days=2)).astimezone(timezone.utc)
```

Comment:
could we do this instead
```suggestion
            datetime.max.replace(tzinfo=timezone.utc)
```